src/utils/querybuilder.util.js

```diff
@@ -514,17 +516,98 @@ const quantityQueryBuilder = function ({ target, field }) {
                 qB[`${field}.value`] = { $gte: num };
                 break;
             case 'ne':
-                qB[`${field}.value`] = { $ne: num };
+                if (strNum.indexOf('e') > 0) {
+                    range = calcRangeSN(numStr);
+                } else {
+                    range = calcRange(strNum);
+                }
+                num = Number(num);
+                qB[`${field}.value`] = { $ne: { $gte: Number((num - range.offset).toPrecision(range.precn)),
+                        $lt: Number((num + range.offset).toPrecision(range.precn)) }};
                 break;
         }
     } else {
         // no prefixes
-        qB[`${field}.value`] = Number(num);
+       if (num.indexOf('e') > 0) {
+           range = calcRangeSN(num);
+       } else {
+           range = calcRange(num);
+       }
+       num = Number(num);
+       qB[`${field}.value`] = { $gte: Number((num - range.offset).toPrecision(range.precn)),
+                        $lt: Number((num + range.offset).toPrecision(range.precn)) }
     }
 
     return qB;
 };
 
+// return value of range offset and precision
+function calcRange(numStr) {
+    const n = Number(numStr);
+    if (n === 0) return 0.5;
+
+    // Remove leading zeros and the decimal point
+    if (numStr.indexOf('.') !== -1) {
+        // Floating-point number
+        numStr = numStr.replace(/^0+/g, ''); // Remove leading zeros
+        numStr = numStr.replace('.', ''); // Remove the decimal point
+    } else {
+        // Integer number
+        numStr = numStr.replace(/^0+/, ''); // Remove leading zeros only
+    }
+
+    let numIntDigits = 0;
+    let beforePoint = Math.trunc(n);
+    if (beforePoint !== 0) {
+        beforePoint = beforePoint.toString();
+        numIntDigits = beforePoint.length;
+    }
+
+    const offsetLen = numStr.length;
+    let offset = '0.';
+    for (let i = 0; i < offsetLen; i++) {
+        offset = offset + '0';
+    }
+    offset = offset + '5';
+    offset = Number(offset) * Math.pow(10, numIntDigits);
+    const precn  = numStr.length + 1;
+    return { offset, precn};
+}
+
+
+// return value of range offset for scientific notation numbers
+function calcRangeSN(numStr) {
```

Comment:
For value '1e2' the range currently is {'$gte':50,'$lt':150} but it should be {'$gte':95,'$lt':105}.
Rest all inputs are working fine.
Please add test cases for all different types of inputs.

Reply:
According to R5 https://hl7.org/fhir/R5/search.html#number this is correct
Can you please confirm with Chris, which one will we implement here ? 
As the one in R5 seems consistent.

Reply:
Added kludge for 1e2 case
Added test cases